### ISSUE TITLE
hexpansion header doesn't seem to flash on my badges using zetta eeprom

### ISSUE THREAD

**Issue by thinkl33t** [emfcamp/badge-2024-software#59]
changing line:

```
write_header(
    port, header, addr=addr, addr_len=addr_len, page_size=header.eeprom_page_size
)
```

to
```
i2c.writeto(addr, bytes([0, 0]) + header.to_bytes())
```

fixes it though, I'm not familiar enough with i2c to know why though

**Comment by hairymnstr** [comment]
This work around will fix some bugs in the code (fixed pending merge) but only for users of the ZD24C64A EEPROM.

**Comment by hairymnstr** [comment]
Fixed in 1.8.0 the code checks the address length and writes the correct number of address bytes for the EEPROM when writing header chunks.
